Modules/III-k6-Intermediate/02-Dynamic-correlation-in-k6.md

```diff
@@ -4,18 +4,18 @@ In the previous section, you learned [different approaches to debugging a script
 
 ## What is correlation?
 
-Correlation is the process of extracting a value from a previous HTTP response and using that value in the next HTTP request.
+[Data correlation](https://k6.io/docs/misc/glossary/#data-correlation) is the process of extracting a value from a previous HTTP response and using that value in the next HTTP request.
 
 The request flow of a user accessing an application might look like this:
 - Step 1. HTTP GET request for *My Messages* page.
 - Step 2. HTTP POST request with username and password
 
-In the previous section, the script attempted to log in immediately (skipping Step 1 and going straight to Step 2), but after applying some debugging techniques, you discovered that the response code is an HTTP 403 Forbidden, with `error: invalid csrf token` in the response body.
+In the [previous section](./01-How-to-debug-k6-load-testing-scripts.md), the script attempted to log in immediately (skipping Step 1 and going straight to Step 2), but after applying some debugging techniques, you discovered that the response code is an HTTP 403 Forbidden, with `error: invalid csrf token` in the response body.
 
 To correct this error and accurately mimic user behavior, the script must:
 - Save the response to the request in Step 1.
-- Extract the csrf token from Step 1.
-- Pass the csrf token when requesting Step 2.
+- Extract the [Cross-Site Request Forgery (CSFR)](https://owasp.org/www-community/attacks/csrf) token from Step 1.
```

Comment:
```suggestion
- Extract the [Cross-Site Request Forgery (CSFR)](https://owasp.org/www-community/attacks/csrf) token from Step 1.
```
Is this the first place that CSFR appears? I think we might want to link it there, or not link unless it's necessary here. Generally, I try to avoid links in direct procedures, since links naturally call to themselves, which may divide the reader's attention.

Willing to change my opinion though if the info is particularly useful here.

Reply:
I think it was the first appearance, and for newcomers it's good to explain what this is (unless we had a glossary)